versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/persist/ObjIdHasher.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.storage.common.persist;
+
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.Map;
+import java.util.UUID;
+
+public interface ObjIdHasher {
+  static ObjIdHasher objIdHasher(String typeName) {
+    return new ObjIdHasherImpl().hash(typeName);
+  }
+
+  static ObjIdHasher objIdHasher(Enum<?> enumValue) {
+    return objIdHasher(enumValue.name());
```

Comment:
It could be null, in theory. But yea - mostly to keep it consistent with the other functions.

This function is an abbreviation/convenience for the existing `enumValue.name()` hasher invocations.
Not a fan of including an implementation detail (class name) into the hash.